src/mailbox-state.c

```diff
@@ -777,8 +790,15 @@ int mailbox_state_set_permanent_flags(struct mailbox_view *view,
 		if (*atom == '\\') {
 			if (strcmp(atom, "\\*") == 0)
 				view->keywords_can_create_more = TRUE;
-			else if (mail_flag_parse(atom + 1) == 0)
+			else if ((flags = mail_flag_parse(atom + 1)) == 0)
+				return -1;
+			if ((flags & MAIL_RECENT) != 0 && imap4rev2_enabled) {
+				i_error("Unexpectedly trying to set a RECENT"
+					 "flag, while IMAP4REV2 is enabled");
```

Comment:
Similarly this `Mailbox PERMANENTFLAGS response contains \Recent flag, while IMAP4REV2 is enabled